kyuubi-rest-client/src/main/java/org/apache/kyuubi/client/RetryableRestClient.java

```diff
@@ -57,7 +57,7 @@ public static IRestClient getRestClient(List<String> uris, RestClientConf conf)
             client);
   }
 
-  private void newRestClient() {
+  private synchronized void newRestClient() {
```

Comment:
the client may be used in multi threads

Reply:
it will not be used in multi threads.. we can improve the class docs, each thread should hold its own instance of RetryableRestClient

Reply:
In practice, an HTTP-based client should be thread-safe.